src/middlewared/middlewared/plugins/system_vendor/vendor.py

```diff
@@ -11,18 +11,25 @@
 
 class VendorService(Service):
 
+    class Config:
+        namespace = 'system.vendor'
+        cli_private = True
+
     @api_method(VendorNameArgs, VendorNameResult, private=True)
     def name(self) -> str | None:
         try:
             with open(SENTINEL_FILE_PATH, 'r') as file:
-                if contents := file.read():
-                    return json.loads(contents).get('name')
+                return json.load(file).get('name') or None  # Don't return an empty string.
         except (FileNotFoundError, json.JSONDecodeError):
-            return None
+            pass
+        except Exception:
+            self.logger.error('Unexpected error while reading %r', SENTINEL_FILE_PATH, exc_info=True)
 
     @api_method(UnvendorArgs, UnvendorResult, private=True)
     def unvendor(self):
         try:
             os.remove(SENTINEL_FILE_PATH)
         except FileNotFoundError:
-            return None
+            pass
+        except Exception:
```

Comment:
Maybe better to catch `OSError` along with the one we are already handling and just leave it at that as highly unlikely something else will come here?

Reply:
The theory behind this unsatisfying exception handling came from a discussion with @yocalebo . The plugin should _never_ raise any exceptions. The safest way to guarantee this is to catch everything. However it was decided to separate the "expected" exceptions (e.g. `FileNotFoundError`) from the highly unlikely catch-all.

Reply:
Sure, if @yocalebo agrees we are good 👍 My perception is that in such cases we should always try to catch all the possible exceptions which can arise and mark them appropriately - so if by any chance something out of the blue arises and it actually needs to be fixed, it would never be raised because we were catching everything

Reply:
Also if we were doing something really complex which could result in many exceptions being raised, then probably yes i would prefer using base exception but still log it in that case so we know something went south and if it needs to be fixed we can have it fixed